src/pudl/transform/ferc1.py

```diff
@@ -1693,6 +1885,82 @@ def plants_steam(ferc1_raw_dfs, ferc1_transformed_dfs):
+    """Transforms FERC Form 1 plant_steam data for loading into PUDL Database.
+
+    Args:
+        ferc1_dbf_raw_dfs: Each entry in this dictionary of DataFrame objects
+            corresponds to a table from the  FERC Form 1 DBC database.
+        ferc1_xbrl_raw_dfs: Each entry in this dictionary of DataFrame objects
+            corresponds to a table from the  FERC Form 1 XBRL database.
+        ferc1_transformed_dfs: A dictionary of DataFrames to be transformed.
+    """
+    plants_steam_dbf = pre_concat_dbf_plants_steam(ferc1_dbf_raw_dfs)
+    plants_steam_xbrl = pre_concat_xbrl_plants_steam(ferc1_xbrl_raw_dfs)
+
+    plants_steam_combo = (
+        pd.concat([plants_steam_dbf, plants_steam_xbrl])
+        .pipe(pudl.helpers.simplify_strings, ["plant_name_ferc1"])
+        .pipe(
+            pudl.helpers.cleanstrings,
+            ["construction_type", "plant_type"],
+            [CONSTRUCTION_TYPE_STRINGS, PLANT_KIND_STRINGS],
+            unmapped=pd.NA,
+        )
+        .pipe(
+            pudl.helpers.oob_to_nan,
+            cols=["construction_year", "installation_year"],
+            lb=1850,
+            ub=max(DataSource.from_id("ferc1").working_partitions["years"]) + 1,
+        )
+        .assign(
+            capex_per_mw=lambda x: 1000.0 * x.capex_per_kw,
+            opex_per_mwh=lambda x: 1000.0 * x.opex_per_kwh,
+            net_generation_mwh=lambda x: x.net_generation_kwh / 1000.0,
+        )
+        .drop(columns=["capex_per_kw", "opex_per_kwh", "net_generation_kwh"])
+        .pipe(convert_cols_dtypes, data_source="ferc1")
+    )
+    return plants_steam_combo
+
+
+def pre_concat_dbf_plants_steam(ferc1_dbf_raw_dfs):
+    """Modifications of the dbf plants_steam_ferc1 table before concat w/ xbrl."""
+    plants_steam_dbf = rename_table(
+        ferc1_dbf_raw_dfs["plants_steam_ferc1"],
+        table_name="plants_steam_ferc1",
+        source="dbf",
+    ).pipe(_clean_cols, "f1_steam")
+    return plants_steam_dbf
+
+
+def pre_concat_xbrl_plants_steam(ferc1_xbrl_raw_dfs):
+    """Modifications of the xbrl plants_steam_ferc1 table before concat w/ xbrl."""
+    plants_steam_xbrl = (
+        merge_instant_and_duration_tables(
+            duration=ferc1_xbrl_raw_dfs[
+                "steam_electric_generating_plant_statistics_large_plants_402_duration"
+            ],
+            instant=ferc1_xbrl_raw_dfs[
+                "steam_electric_generating_plant_statistics_large_plants_402_instant"
+            ],
+            merge_on=["PlantNameAxis", "entity_id"],
+        )
+        .pipe(
+            rename_table,
+            table_name="plants_steam_ferc1",
+            source="xbrl",
+        )
+        .assign(report_year=lambda x: pd.to_datetime(x.start_date).dt.year)
+        .pipe(
+            assign_record_id_xbrl,
+            table_name="plants_steam_ferc1",
+            idx_cols=["plant_name_ferc1", "entity_id", "report_year"],
+        )
+    )
+    return plants_steam_xbrl
```

Comment:
It seems like this is likely to be the outline of the steps for many (all?) of the tables to get them to where they can be concatenated with the DBF data. If that's true, then how can we separate the actions from the parameters that define exactly what they do, so that the code is just in one place, and all we're storing on a per-table basis is data / metadata or potentially a `Callable` (function) that's either passed in or a class method if necessary?

* Query the XBRL derived DB and maybe do one join to bring in the entity ID? Can this be done knowing only the XBRL DB table name that corresponds to the requested PUDL FERC 1 table?
* Merging the instantaneous & time period values: Are all of the tables we're building based on both types of data? If not, can we have one function that does the right thing when both kinds of data aren't present? Does this action need to be parameterized with some kind of input beyond the name of the table, or is it the same every time? Rather than having the names of the instant/duration tables hard coded inside every different transform function, can they be passed in? Or can they be something that's looked up in the metadata structures, given the ID of the table we're transforming? It seems like there's `instant_table`, `duration_table`, and `merge_on`. How would we record this upstream dependency in Dagster?
* Rename the columns, looking up the from/to column names based on the XBRL/DBF source and the name of the table.
* Do all of the tables have a report year?
* Can we make `assign_record_id()` another function like rename, that applies to either XBRL or DBF, and switches its behavior appropriately? This would mean pulling that action out of `_clean_cols()` for the old data.